pkg/deploy/api/validation/validation.go

```diff
@@ -39,6 +39,9 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	if spec.Template == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
 	} else {
+		originalContainerImageNames := getContainerImageNames(spec.Template)
+		defer setContainerImageNames(spec.Template, originalContainerImageNames)
```

Comment:
How about moving the `getContainerImageNames` and `defer setContainerImageNames` *inside* the `handleEmptyImageReferences` call, where they get called in succession / one right after the other, once we determine the empty image field / ICT present condition exists.  I think that would achieve both the safety concern as well avoid the use of defer if the emtpy image field / ICT present condition does NOT exist.